app/src/main/java/com/example/drivescience/OnClearFromRecentService.java

```diff
@@ -0,0 +1,74 @@
+package com.example.drivescience;
+
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.app.PendingIntent;
+import android.app.Service;
+import android.content.Intent;
+import android.os.Build;
+import android.os.IBinder;
+import android.util.Log;
+
+import androidx.annotation.Nullable;
+import androidx.core.app.NotificationCompat;
+import androidx.core.app.NotificationManagerCompat;
+
+public class OnClearFromRecentService extends Service {
+
+    static int NOTIFICATION_ID = 1236;
+
+    @Nullable
+    @Override
+    public IBinder onBind(Intent intent) {
+        return null;
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        NotificationManagerCompat notificationManager = NotificationManagerCompat.from(this);
+        notificationManager.cancel(NOTIFICATION_ID);
+
+        createNotification();
+        return START_NOT_STICKY;
+    }
+
+    @Override
+    public void onTaskRemoved(Intent rootIntent) {
+        createNotification();
+
+        stopSelf();
+    }
+
+    private void createNotification() {
+        createNotificationChannel();
+
+        Intent intent = new Intent(this, MainActivity.class);
+        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+        PendingIntent pendingIntent = PendingIntent.getActivity(this, 0, intent, 0);
+
+        NotificationCompat.Builder builder = new NotificationCompat.Builder(this, "1234")
```

Comment:
What is "1234"? That _feels_ like dummy test data, but I'm guessing it's something to do with android internals?

Reply:
Or should this be using the `NOTIFICATION_ID` you defined up above?

Reply:
This needed to match the channel ID actually. Updated :+1: